src/constructs/core/mappings.ts

```diff
@@ -1,14 +1,46 @@
 import { CfnMapping } from "@aws-cdk/core";
 import type { Stage } from "../../constants";
+import type { AppIdentity } from "./identity";
 import type { GuStack } from "./stack";
 
-export interface GuStageDependentValue<T extends string | number | boolean> {
+export type GuMappingValue = string | number | boolean;
+
+export interface GuStageMappingValue<T extends GuMappingValue> extends AppIdentity {
   variableName: string;
   stageValues: Record<Stage, T>;
 }
 
-export class GuStageMapping extends CfnMapping {
+export class GuStageMapping {
+  private readonly scope: GuStack;
+
+  // the key in this Map is an app string
+  private mappings: Map<string, CfnMapping>;
+
   constructor(scope: GuStack) {
-    super(scope, "stage-mapping");
+    this.scope = scope;
+    this.mappings = new Map<string, CfnMapping>();
+  }
+
+  private getCfnMapping(app: string): CfnMapping {
+    if (!this.mappings.has(app)) {
+      const emptyCfnMapping = new CfnMapping(this.scope, app);
+      this.mappings.set(app, emptyCfnMapping);
+      return emptyCfnMapping;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- `this.mappings.has` above is `true`, so we can safely assert non-null
+    return this.mappings.get(app)!;
+  }
```

Comment:
This is the main change in this PR - we now maintain a [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) of `app -> CfnMapping`.

The rest of this PR is "noise" identified via the compiler and tests.